src/routes/route.py

```diff
@@ -91,11 +101,61 @@ def create_route_objects(
     total_population = _get_total_population(population_data, cars_per_person)
     result = []
     departure_times = _departure_times(total_population, start, end)
-    for route_path in tqdm(list_of_paths):
+    for origin_point in tqdm(origin_to_paths.keys()):
+        paths = origin_to_paths[origin_point]
         num_people_on_route = _get_num_people_on_route(
-            route_path, population_data, cars_per_person
+            origin_point, population_data, cars_per_person
         )
-        selected, departure_times = np.split(departure_times, [num_people_on_route])
-        route_object = Route(route_path, num_people_on_route, list(selected))
-        result.append(route_object)
+        number_of_diverse_routes = len(paths)
+        if number_of_diverse_routes < 1:
+            logging.warning(
+                f"Origin point {origin_point} has no routes to safety. Skipping."
+            )
+            continue
+        elif number_of_diverse_routes == 1 or num_people_on_route < len(paths):
+            route_path = paths[0]
+            route_object, departure_times = _create_route_object(
+                departure_times=departure_times,
+                path=route_path,
+                num_people_on_route=num_people_on_route,
+            )
+            result.append(route_object)
+        else:
+            continue
     return result
+
+
+def population_diversify_route_math(
+    num_people_on_route: int, paths: list[path], departure_times: NDArray[np.int_]
+) -> Tuple[list[Route], NDArray[np.int_]]:
+    """
+    Diversifies the routes for a given number of people on a route.
+    :param num_people_on_route: The number of people on the route.
+    :param paths: A list of paths.
+    :param departure_times: A list of departure times.
+    :return: A list of Route objects.
+    """
+    num, remainder = divmod(num_people_on_route, len(paths))
+    result = []
+    for p in paths:
+        route, departure_times = _create_route_object(
+            departure_times=departure_times, path=p, num_people_on_route=num
+        )
+        result.append(route)
+
```

Comment:
Why aren't we adding the remainder here? I see you made another method, but why not just do it from here during the creation and not afterwards?

Reply:
Neither add_remainder nor population_diversify_route_math is called from this file. Why not keep the logic in here?